ioncore/netwm.c

```diff
@@ -101,6 +106,10 @@ void netwm_init_rootwin(WRootWin *rw)
     xwindow_set_utf8_property(rw->dummy_win, atom_net_wm_name, p, 1);
 }
 
+void netwm_set_frame_extents(WClientWin* cwin, long* extents) {
+    XChangeProperty(ioncore_g.dpy, cwin->win, atom_net_frame_extents,
+                    XA_CARDINAL, 32, PropModeReplace, (uchar*)extents, 4);
+}
```

Comment:
This is not a very safe API: The caller needs to know that they have to pass 4 values in `extents`. I'd rather make this explicit, which also allows hiding the gotcha that a protocol-level `CARD32` is represented a `long` in the xlib ABI, no matter the actual `sizeof(long)`, while the actual `CARD32` type varies in size:

```suggestion
void netwm_set_frame_extents(WClientWin *cwin, CARD32 left, CARD32 right, CARD32 top, CARD32 bottom)
{
    unsigned long data[4];

    data[0] = left;
    data[1] = right;
    data[2] = top;
    data[3] = bottom;

    XChangeProperty(ioncore_g.dpy, cwin->win, atom_net_frame_extents,
                    XA_CARDINAL, 32, PropModeReplace, (uchar*)data, 4);
}
```

Reply:
I agree it would be better to pass in the 4 values individually (or in a dedicated data structure). I think I'd prefer to pass them in as regular `int`s and do the conversion to `long`/`CARD32` inside the function, though. Amended, WDYT?

Reply:
That's fine, I guess that's more consistent with the rest of the codebase anyway.